### DISCUSSION TITLE
How to deploy offline

### DISCUSSION THREAD

**Discussion by feranwq** [AbdBarho/stable-diffusion-webui-docker#357] [General]
Is there an offline deployment guide? my network is limited, I tried to create the images in GCP vm,
build services/AUTOMATIC1111/Dockerfile and services/download/Dockerfile to run locally, but I get an error OSError: Can't load tokenizer for 'openai/clip-vit-large-patch14'. I'm not sure what else I need after that, so I'd like to have an offline deployment guide

**Comment by davedavis** [comment]
This repo downloads the needed models at both build time and run time dynamically. So curious, how do you expect to get all the models? 

You could download them all yourself and upload them to your VM? Is that what you're looking to do?

**Reply by feranwq** [reply]
Yes, I have downloaded the models from download/Dockerfile, but when I start auto ui it says there is no openai/clip-vit-large-patch14, I want to know what are the runtime models and how to download and use them manually?

**Comment by davedavis** [comment]
Don't think there's a guide, but it's not just the standard diffusion models you need. There's a bunch of other models needed. 
So [here](https://huggingface.co/openai/clip-vit-large-patch14) is the one you're looking for but you're going to need a lot more. The links are in the links file [here](https://github.com/AbdBarho/stable-diffusion-webui-docker/blob/master/services/download/links.txt).

You might be better off setting up locally then uploading to your VPS.

**Reply by feranwq** [reply]
thank you very much for your help, after I downloaded openai/clip-vit-large-patch14, I mounted it with . /data/openai:/stable-diffusion-webui/openai, now it works fine, I added some additional models, thanks again!
